code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm

```diff
@@ -0,0 +1,420 @@
+	else
+		var/mob/living/simple_animal/hostile/poison/terror_spider/T = killtarget
+		if (T.ckey)
+			// living player
+			ts_ckey_blacklist += T.ckey
+		visible_message("<span class='danger'> [src] grabs hold of [T] and tears them limb from limb! </span>")
+		T.death()
+		T.gib()
+		regen_points += regen_points_per_kill
+
+/mob/living/simple_animal/hostile/poison/terror_spider/queen/verb/LayQueenEggs()
+	set name = "Lay Queen Eggs"
+	set category = "Spider"
+	set desc = "Grow your brood."
+	if (canlay < 1)
+		var/remainingtime = round(((spider_lastspawn + spider_spawnfrequency) - world.time) / 10,1)
+		if (remainingtime > 0)
+			to_chat(src, "Too soon to attempt that again. Wait another " + num2text(remainingtime) + " seconds.")
+		else
+			to_chat(src, "Too soon to attempt that again. Wait just a few more seconds...")
+		return
+	var/list/eggtypes = list("red - assault","gray - ambush", "green - nurse", "black - poison","purple - guard")
+	if (canlay >= 12)
+		eggtypes |= "MOTHER - ELITE HORROR"
+		eggtypes |= "PRINCE - ELITE WARRIOR"
+	var/eggtype = input("What kind of eggs?") as null|anything in eggtypes
+	if (!(eggtype in eggtypes))
+		to_chat(src, "Unrecognized egg type.")
+		return 0
+	if (eggtype == "MOTHER - ELITE HORROR" || eggtype == "PRINCE - ELITE WARRIOR")
+		if (canlay < 12)
+			to_chat(src, "Insufficient strength. It takes as much effort to lay one of those as it does to lay 12 normal eggs.")
+		else
+			if (eggtype == "MOTHER - ELITE HORROR")
+				canlay -= 12
+				DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/mother,1,0)
+			else if (eggtype == "PRINCE - ELITE WARRIOR")
+				canlay -= 12
+				DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/prince,1,0)
+		return
+	var/numlings = 1
+	if (canlay >= 5)
+		numlings = input("How many in the batch?") as null|anything in list(1,2,3,4,5)
+	else if (canlay >= 3)
+		numlings = input("How many in the batch?") as null|anything in list(1,2,3)
+	else if (canlay == 2)
+		numlings = input("How many in the batch?") as null|anything in list(1,2)
+	if (eggtype == null || numlings == null)
+		to_chat(src, "Cancelled.")
+		return
+	//spider_lastspawn = world.time // don't think we actually need this, if queen is laying manually canlay controls her rate.
+	canlay -= numlings
+	if (eggtype == "red - assault")
+		DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/red,numlings,1)
+	else if (eggtype == "gray - ambush")
+		DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/gray,numlings,1)
+	else if (eggtype == "green - nurse")
+		DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/green,numlings,1)
+	else if (eggtype == "black - poison")
+		DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/black,numlings,1)
+	else if (eggtype == "purple - guard")
+		DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/purple,numlings,0)
```

Comment:
No. The last 0 controls whether the spiderlings ventcrawl before reaching maturity. Not how many there are. In this case, the 0 prevents purple spiders (whose entire job is guarding the queen) from moving away from her before they mature.